src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -859,13 +859,14 @@ VideoDecoder::DecodedOutput VideoDecoder::convertAVFrameToDecodedOutput(
     convertAVFrameToDecodedOutputOnCPU(
         rawOutput, output, preAllocatedOutputTensor);
   } else if (streamInfo.options.device.type() == torch::kCUDA) {
-    // TODO: handle pre-allocated output tensor
+    // TODO: we should fold preAllocatedOutputTensor into RawDecodedOutput.
```

Comment:
Nit: move this TODO outside of this else/if block (just on top of it?), because this applies to CPU as well, not just to the CUDA branch. We may also want to open an issue?

Reply:
Done